synapse/lib/lmdb.py

```diff
@@ -0,0 +1,443 @@
+import struct
+import itertools
+
+import lmdb  # used for type resolution
+
+import synapse.lib.msgpack as s_msgpack
+
+STOR_FLAG_NOINDEX = 0x0001      # there is no byprop index for this prop
+STOR_FLAG_MULTIVAL = 0x0002     # this is a multi-value prop
+STOR_FLAG_DEFVAL = 0x0004       # Only set this if it doesn't exist
```

Comment:
I'm unclear what the semantics of defval is. Does that mean the value is the default value for that property? If so, I think there's some missing logic when setting the value.